generators/aws/lib/aws.js

```diff
@@ -1,25 +1,35 @@
 'use strict';
 var S3 = require('./s3.js'),
     Rds = require('./rds.js'),
-    chalk = require('chalk'),
+    shelljs = require('shelljs'),
+    os = require('os'),
     Eb = require('./eb.js');
 
 var Aws, generator;
 
-var AwsFactory = module.exports = function AwsFactory(generatorRef) {
+var AwsFactory = module.exports = function AwsFactory(generatorRef, cb) {
+    generator = generatorRef;
     try {
         Aws = require('aws-sdk');
-        generator = generatorRef;
+        cb();
     } catch (e) {
-        generator.env.error(chalk.red(
-            'You don\'t have the AWS SDK installed. Please install it in the JHipster generator directory.\n\n') +
-            chalk.yellow('WINDOWS\n') +
-            chalk.green('cd %USERPROFILE%\\AppData\\Roaming\\npm\\node_modules\\generator-jhipster\n' +
-            'npm install aws-sdk progress node-uuid\n\n') +
-            chalk.yellow('LINUX / MAC\n') +
-            chalk.green('cd /usr/local/lib/node_modules/generator-jhipster\n' +
-            'npm install aws-sdk progress node-uuid')
-        );
+        generator.log('Installing AWS dependencies into your JHipster folder');
+        var jhipsterPath = 'node_modules/generator-jhipster';
+        var skipClient = generator.config.get('skipClient');
+        // use the global JHipster for apps with no client-side code
+        if (skipClient) {
+            var prefix = shelljs.exec('npm config get prefix', {silent:true}).trim();
+            if (os.platform() === 'win32') {
```

Comment:
The global node_modules path has an extra `/lib/` on Unix compared to Windows.  ShellJS does not handle that.